src/main/java/org/jruby/ext/openssl/OpenSSL.java

```diff
@@ -255,27 +255,27 @@ static void warn(final ThreadContext context, final IRubyObject msg) {
     public static String javaVersion(final String def) {
         final String javaVersionProperty =
                 SafePropertyAccessor.getProperty("java.version", def);
-        if ( javaVersionProperty == "0" ) return "1.7.0"; // Android
+        if (javaVersionProperty.equals("0")) return "1.7.0"; // Android
         return javaVersionProperty;
     }
 
     static boolean javaVersion6(final boolean atLeast) {
-        final int gt = "1.6".compareTo( javaVersion("0.0").substring(0, 3) );
+        final int gt = new Version("1.6").compareTo(new Version(javaVersion("0.0")));
         return atLeast ? gt <= 0 : gt == 0;
     }
 
     static boolean javaVersion7(final boolean atLeast) {
-        final int gt = "1.7".compareTo( javaVersion("0.0").substring(0, 3) );
+        final int gt = new Version("1.7").compareTo(new Version(javaVersion("0.0")));
         return atLeast ? gt <= 0 : gt == 0;
     }
 
     static boolean javaVersion8(final boolean atLeast) {
-        final int gt = "1.8".compareTo( javaVersion("0.0").substring(0, 3) );
+        final int gt = new Version("1.8").compareTo(new Version(javaVersion("0.0")));
         return atLeast ? gt <= 0 : gt == 0;
     }
 
     static boolean javaVersion9(final boolean atLeast) {
-        final int gt = "9".compareTo( javaVersion("0").substring(0, 1) );
+        final int gt = new Version("9").compareTo(new Version(javaVersion("0.0")));
         return atLeast ? gt <= 0 : gt == 0;
     }
```

Comment:
maybe also inner `static class Version implements ...` here, 
prefer not to have 'public' utils (internally) used in one place